src/PotentialEnergyEquationTerms.jl

```diff
@@ -0,0 +1,113 @@
+(:T, :S)
+
+julia> eos = TEOS10EquationOfState()
+BoussinesqEquationOfState{Float64}:
+    ├── seawater_polynomial: TEOS10SeawaterPolynomial{Float64}
+    └── reference_density: 1020.0
+
+julia> buoyancy = SeawaterBuoyancy(equation_of_state=eos)
+SeawaterBuoyancy{Float64}:
+├── gravitational_acceleration: 9.80665
+└── equation_of_state: BoussinesqEquationOfState{Float64}
+
+julia> model = NonhydrostaticModel(; grid, buoyancy, tracers)
+NonhydrostaticModel{CPU, RectilinearGrid}(time = 0 seconds, iteration = 0)
+├── grid: 1×1×100 RectilinearGrid{Float64, Flat, Flat, Bounded} on CPU with 0×0×3 halo
+├── timestepper: QuasiAdamsBashforth2TimeStepper
+├── tracers: (T, S)
+├── closure: Nothing
+├── buoyancy: SeawaterBuoyancy with g=9.80665 and BoussinesqEquationOfState{Float64} with ĝ = NegativeZDirection()
+└── coriolis: Nothing
+
+julia> PotentialEnergy(model)
+KernelFunctionOperation at (Center, Center, Center)
+├── grid: 1×1×100 RectilinearGrid{Float64, Flat, Flat, Bounded} on CPU with 0×0×3 halo
+├── kernel_function: gρz_ccc (generic function with 1 method)
+└── arguments: ("KernelFunctionOperation at (Center, Center, Center)", "KernelFunctionOperation at (Center, Center, Center)", "(g=9.80665, ρ₀=1020.0)")
+```
+
+To use a reference density set a constant value for the keyword argument `geopotential_height`
+and pass this the function. For example,
+```jldoctest
+julia> using Oceananigans, SeawaterPolynomials.TEOS10;
+
+julia> using Oceanostics.PotentialEnergyEquationTerms: PotentialEnergy;
+
+julia> grid = RectilinearGrid(size=100, z=(-1000, 0), topology=(Flat, Flat, Bounded));
+
+julia> tracers = (:T, :S);
+
+julia> eos = TEOS10EquationOfState();
+
+julia> buoyancy = SeawaterBuoyancy(equation_of_state=eos);
+
+julia> model = NonhydrostaticModel(; grid, buoyancy, tracers);
+
+julia> geopotential_height = 0; # density variable will be σ₀
+
+julia> PotentialEnergy(model; geopotential_height)
+KernelFunctionOperation at (Center, Center, Center)
+├── grid: 1×1×100 RectilinearGrid{Float64, Flat, Flat, Bounded} on CPU with 0×0×3 halo
+├── kernel_function: gρz_ccc (generic function with 1 method)
+└── arguments: ("KernelFunctionOperation at (Center, Center, Center)", "KernelFunctionOperation at (Center, Center, Center)", "(g=9.80665, ρ₀=1020.0)")
+```
+"""
+@inline function PotentialEnergy(model; geopotential_height = model_geopotential_height(model), location = (Center, Center, Center))
+
+    validate_location(location, "PotentialEnergy")
+    validate_buoyancy(model.buoyancy)
+
+    grid = model.grid
+    ρ = seawater_density(model; geopotential_height)
+    Z = model_geopotential_height(model)
+    parameters = (g = model.buoyancy.model.gravitational_acceleration,
+                  ρ₀ = model.buoyancy.model.equation_of_state.reference_density)
+
+    return KernelFunctionOperation{Center, Center, Center}(gρz_ccc, grid, ρ, Z, parameters)
+end
+
+@inline gρz_ccc(i, j, k, grid, ρ, Z, p) = (p.g / p.ρ₀) * ρ[i, j, k] * Z[i, j, k]
```

Comment:
One possibility is to implicitly call $g\rho/\rho_0=g\prime$, that way the units work out:

```suggestion
@inline g′z_ccc(i, j, k, grid, ρ, Z, p) = (p.g / p.ρ₀) * ρ[i, j, k] * Z[i, j, k]
```

Reply:
I'm also fine with naming it `gρ₀⁻¹ρz_ccc` like you said, or  `gρz_ρ₀_ccc`, and let the underscore implicitly mean division. As long as it can be understood as a representation of the operation, I'm happy :)

Reply:
I like `g′z_ccc` because it is the most compact.